### DISCUSSION TITLE
Is it normal that build time increase to 1.46m from 10s when set newTreeshaking to true?

### DISCUSSION THREAD

**Discussion by jayjliang** [web-infra-dev/rsbuild#1814] [Q&A]
![Uploading image.png…]()

**Comment by chenjiahan** [comment]
The image is broken.

This should be a bug, can you create an issue in the Rspack repo?  https://github.com/web-infra-dev/rspack/issues

**Reply by jayjliang** [reply]
sorry, the project is private and normal project don't have this problem, is there any tools that can analyse the build time

**Reply by chenjiahan** [reply]
Sure, you can use Rsdoctor to analyze the build time: https://github.com/web-infra-dev/rsdoctor

And there is a profiling chapter in the rspack dev guide: https://web-infra-dev.github.io/rspack-dev-guide/profiling/intro.html#profiling

**Comment by jayjliang** [comment]
rsdoctor can't help me find the key problem
<img width="531" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/9363437/8a026e63-4353-498f-ba78-2896b2da3ecb">
<img width="1664" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/9363437/fc342833-12df-48d5-b4df-b125eea38364">
<img width="992" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/9363437/bdbd1ded-8c50-4aca-bcd1-33d4cce7230b">
i guess it's releated to newTreeshaking, i try to use rspack profilling, the nodejs profiling is normal, how can i get tracing data using rsbuild? the export rspack config is useless

**Comment by 9aoy** [comment]
rspack profile can also be used in rsbuild , you can use `RSPACK_PROFILE=ALL rsbuild build` to get more rspack  build performance profiles.
If this still doesn't help you locate the problem, maybe you can try to provide a minimal reproducible demo.

https://www.rspack.dev/guide/profile.html

**Reply by jayjliang** [reply]
<img width="1728" alt="image" src="https://github.com/web-infra-dev/rsbuild/assets/9363437/d42f7677-385a-40f6-8b63-fb6c74d2bc5d">
the longest slice is `plugin:finish_modules` as above

**Reply by jayjliang** [reply]
[logging.json](https://github.com/web-infra-dev/rsbuild/files/14617400/logging.json)
logging.json has been attached, tracing json (93MB) is too big to upload

**Reply by jayjliang** [reply]
what's more, i really want to provide a minimal reproducible demo, but this project has thousands of private dependencies
if i turn off `newTreeshaking`, there will be [circular dependency problem](https://github.com/web-infra-dev/rspack/issues/5928)

**Comment by jayjliang** [comment]
finally，[#5123](https://github.com/web-infra-dev/rspack/issues/5123) seems the same problem

**Comment by chenjiahan** [comment]
It is recommended to only enable new tree shaking in production build at this stage, see https://github.com/web-infra-dev/rsbuild/pull/1839

**Reply by jayjliang** [reply]
if i turn off newTreeshaking, there will be [circular dependency problem](https://github.com/web-infra-dev/rspack/issues/5928)
